File: frontend/src/types/incident.ts
export type IncidentStatus = 'open' | 'acknowledged' | 'in_progress' | 'resolved';

export type IncidentPriority = 'low' | 'medium' | 'high' | 'critical';

export interface Incident {
  id: string;
  property_id: string;
  incident_type: string;
  title: string;
  description: string;
  status: IncidentStatus;
  priority: IncidentPriority;
  reported_by: string | null;
  assigned_to: string | null;
  created_at: string;
  updated_at: string;
  resolved_at: string | null;
}

File: backend/app/domain/entities/incident.py
from __future__ import annotations

from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel


class IncidentStatus(str, Enum):
    """Lifecycle status of an incident."""

    OPEN = "OPEN"
    ACKNOWLEDGED = "ACKNOWLEDGED"
    IN_PROGRESS = "IN_PROGRESS"
    RESOLVED = "RESOLVED"


class IncidentPriority(str, Enum):
    """Severity / urgency classification for incidents."""

    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"


class Incident(BaseModel):
    """An operational incident reported against a property.

    Incidents progress through a simple state machine:
    OPEN -> ACKNOWLEDGED -> IN_PROGRESS -> RESOLVED.
    """

    id: UUID
    property_id: UUID
    incident_type: str
    title: str
    description: str
    status: IncidentStatus
    priority: IncidentPriority
    reported_by: UUID | None = None
    assigned_to: UUID | None = None
    created_at: datetime
    updated_at: datetime
    resolved_at: datetime | None = None
